### PR TITLE
Fixes various typos in the docs

### PR DESCRIPTION
**Pull request by infomiho**
- Went through most (probably all) docs and fixed the obvious typos
- Grammarly has a VS Code extension which made this process quick

### CODE REVIEW

File: docs/pages/SPA-vs-SSR.page.server.mdx [hunk 1]
```diff
@@ -6,7 +6,7 @@ Modern UI frameworks (React/Vue/...) enable a wide range of render modes:
  - Pre-rendering (aka SSG)
  - HTML-only
 
-We compare these render modes with each other and explain which render mode should be used when.
+We compare these render modes with each other and explain when which render mode should be used.
```
Comment (brillout) [inline docs/pages/SPA-vs-SSR.page.server.mdx:9]: How about this?
> We compare render modes and explain when to use each mode.

Comment (infomiho) [reply docs/pages/SPA-vs-SSR.page.server.mdx:9]: I agree it sounds much more natural 👍 I'll update the line.

File: docs/pages/dot-page-dot-client-dot-js.page.server.mdx [hunk 2]
```diff
@@ -3,7 +3,7 @@ Environment: browser.
 The `.page.client.js` file defines the page's browser-side code.
 
 It represents the entire browser-side code. This means that if don't create any `.page.client.js` file, then our app has zero browser-side JavaScript.
```
[the remark this answers is not in the document]
Comment (infomiho) [reply docs/pages/dot-page-dot-client-dot-js.page.server.mdx:5]:
```diff
- This means that if don't create any `.page.client.js` file
+ This means that if we don't create any `.page.client.js` files
```

Going even further 😄

**Comment by brillout** [comment]
💯

**Pull request merged**